src/protocol/errors.rs

```diff
@@ -71,8 +71,20 @@ impl From<Box<dyn error::Error + Send + Sync>> for ProtocolError {
 /// These are related to bad parameters for the protocol, and things like that.
 ///
 /// These are usually more recoverable than other protocol errors.
-#[derive(Debug, Error)]
+#[derive(Debug, Error, Clone, PartialEq, Eq)]
 pub enum InitializationError {
-    #[error("bad parameters: {0}")]
-    BadParameters(String),
+    #[error("bad parameter: {0}")]
+    BadParameter(String),
+    #[error("participant list cannot contain duplicates")]
+    DuplicateParticipants,
+    #[error(
+        "threshold {threshold} cannot be larger than the number of participants {participants}"
+    )]
+    NotEnoughParticipants { threshold: u32, participants: u32 },
```

Comment:
I see that you tried to generalize the error of not having enough participants, and not having enough participants for a given threshold. I would rather keep the separate
```suggestion
    #[error( "participant count cannot be < 2, found: {participants}" )]
    NotEnoughParticipants { participants: u32 },
    #[error(
        "not enough intersecting old/new participants ({participants}) to reconstruct private key for resharing with threshold bigger than old threshold ({threshold})"
    )]
    NotEnoughParticipantsForThreshold { threshold: u32, participants: u32 },
```